### ISSUE TITLE
Diagnostics with GOOGLE_APPLICATION_CREDENTIALS issues

### ISSUE THREAD

**Issue by rokadias** [GoogleCloudPlatform/google-cloud-dotnet#999]
So I initially set up my aspnet core webapi using `gcloud auth` and using my own personal user. The functionality worked within Stackdriver Trace, ErrorReporting, and Stackdriver Logging. I've then proceeded to run the app with the service account and set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable. The StackDriver Logging and ErrorReporting work in this situation, but I no longer see tracing data from the app.

I've ran testIamPolicy on my service account and verified that I have the `cloudtrace.traces.patch` permission. I've even changed my service account to be a project owner so it had the same permission level as my own user, and still haven't seen tracing data received by stackdriver.

An issue that might also be related is that I've also deployed this code to google container engine using a docker image using the same service account with the `GOOGLE_APPLICATION_CREDENTIALS` variable. However, now I don't see Trace, Logging, or Error Reporting data from those docker containers.

**Comment by iantalarico** [comment]
@rokadias I'll give this a test tomorrow morning with a service account to see if I can repo it.  In the mean time what docker image are you using?

**Comment by rokadias** [comment]
`FROM microsoft/dotnet:1.1-sdk-projectjson`

**Comment by iantalarico** [comment]
@rokadias This may be an odd suggestion but how are you viewing your traces?  In the Cloud Console UI?  What time frame are you looking in?

**Comment by rokadias** [comment]
@iantalarico Console UI. I switched to the 1 hour view and was looking for new requests. I basically can unset my service account and see the traces show up in the cloud console. Set the service account back and see lack of traces.

**Comment by iantalarico** [comment]
@rokadias I'm having a hard time repoing the issue.  Do you know what type of buffer you are using?  If you didn't set any you are probably using a timed buffer (which will swallow some exceptions that could kill your process, see #842).  Could you try using no buffer?

To do this when you add trace do the following:
var traceConfig = TraceConfiguration.Create(bufferOptions: BufferOptions.NoBuffer());
services.AddGoogleTrace("project-id", traceConfig);

**Comment by rokadias** [comment]
@iantalarico Looks like now that swallowing of the exceptions is not occurring I can diagnose the problems. The StackDriver Tracing API was not enabled. I enabled the other Apis, but some how missed this one. I don't know why using my personal account got around that issue, but I enabled the API and it fixed the StackDriver Tracing. I'm going to turn off buffering for the moment in my Google Container Engine pods and see if that will expose any exceptions.

The expose exceptions via BufferOptions might be good to have in a troubleshooting area for initial set up and maybe include which APIs should be enabled along with what IAM permissions to have (or links to documentation that describe what's needed).
